src/main/scala/org/constellation/trust/DataGeneration.scala

```diff
@@ -33,30 +34,57 @@ case class TrustNode(id: Int, xCoordinate: Double, yCoordinate: Double, edges: S
     }
   }
 
+  def positiveEdges: Seq[TrustEdge] = edges.filter(_.trust > 0)
+
 }
 
 object DataGeneration {
   val sqrt2: Double = Math.sqrt(2)
 
-  def generateTestData(numNodes: Int = 30): List[TrustNode] = {
+  def randomEdgeLogic(random: Random = new Random())(distance: Double): Boolean =
+    Random.nextDouble() > distance && Random.nextDouble() < 0.5
+
+  def randomEdge(random: Random = new Random())(logic: Double => Boolean = randomEdgeLogic(random))(n: TrustNode, n2: TrustNode) =
+    if (logic(n.id)) {
+      val trustZeroToOne = Random.nextDouble()
+      Some(TrustEdge(n.id, n2.id, 2 * (trustZeroToOne - 0.5), isLabel = true))
+    } else None
+
+  def generateFullyConnectedTestData(numNodes: Int = 30) =
```

Comment:
Test data should be generated in tests only. It shouldn't be mixed with the protocol code.